app/store/reducers/systems.js

```diff
@@ -0,0 +1,32 @@
+import { createSlice } from '@reduxjs/toolkit'
+export const slice = createSlice({
+  name: 'systems',
+  initialState: {
+    loading: false,
+    errorMessage: null,
+    data: {},
+  },
+  reducers: {
+    fetchSystems: (state) => {
+      state.loading = true
+      state.data = {}
+    },
+    fetchSystemsFulfilled: (state, action) => {
+      state.loading = false
+      state.data = action.payload.data
+      state.errorMessage = null
+    },
+    fetchSystemsRejected: (state) => {
+      state.loading = false
+      state.errorMessage = 'Error data list'
```

Comment:
Again update the error message to be more detailed and user friendly. Then add a selector to use this as an error message to the user.